tests/UKHO.ERPFacade.EventAggregation.WebJob.Tests/Helpers/RecordOfSaleSapMessageBuilderTests.cs

```diff
@@ -313,6 +318,87 @@ public void WhenTransactionTypeIsMaintainHoldings_ThenReturns_SomeFieldsEmptyInS
+            var jsonData2 = JsonConvert.DeserializeObject<RecordOfSaleEventPayLoad>(jsonForMigrateNewLicence2);
+
+            List<RecordOfSaleEventPayLoad> rosMigrateNewLicenceData = new()
+            {
+                jsonData1!, jsonData2!
+            };
+
+            string correlationId = "123-abc-456-xyz-333";
+            string sapReqXml = TestHelper.ReadFileData("ERPTestData\\MigrateNewLicencePayloadTest.xml");
+
+            XmlDocument soapXml = new();
+            soapXml.LoadXml(RosSapXmlFile);
+
+            A.CallTo(() => _fakeFileSystemHelper.IsFileExists(A<string>.Ignored)).Returns(true);
+            A.CallTo(() => _fakeXmlHelper.CreateXmlDocument(A<string>.Ignored)).Returns(soapXml);
+            A.CallTo(() => _fakeXmlHelper.CreateXmlPayLoad(A<SapRecordOfSalePayLoad>.Ignored)).Returns(sapReqXml);
+
+            var result = _fakeRecordOfSaleSapMessageBuilder.BuildRecordOfSaleSapMessageXml(rosMigrateNewLicenceData, correlationId);
+
+            result.Should().BeOfType<XmlDocument>();
+
+            var actionItem = result.SelectSingleNode(XpathZAddsRos);
+            actionItem.ChildNodes.Count.Should().Be(1);
+            actionItem.ChildNodes[0].ChildNodes.Count.Should().Be(21);
+
+            var licNo = result.SelectSingleNode(XpathLicNo);
+            var fleet = result.SelectSingleNode(XpathFleet);
+            licNo.InnerXml.Should().BeEmpty();
+            fleet.InnerXml.Should().BeEmpty();
+
+            var prodItem = result.SelectSingleNode(XpathProd);
+            prodItem.ChildNodes.Count.Should().Be(4);
+            prodItem.ChildNodes[0].ChildNodes.Count.Should().Be(5);
+
+            var repeat = result.SelectSingleNode(XpathRepeat);
+            repeat.InnerXml.Should().BeEmpty();
+
+            A.CallTo(_fakeLogger).Where(call => call.Method.Name == "Log"
+            && call.GetArgument<LogLevel>(0) == LogLevel.Information
+            && call.GetArgument<EventId>(1) == EventIds.CreatingRecordOfSaleSapPayload.ToEventId()
+            && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "Creating the record of sale SAP Payload. | _X-Correlation-ID : {_X-Correlation-ID}").MustHaveHappenedOnceExactly();
+
+            A.CallTo(_fakeLogger).Where(call => call.Method.Name == "Log"
+            && call.GetArgument<LogLevel>(0) == LogLevel.Information
+            && call.GetArgument<EventId>(1) == EventIds.CreatedRecordOfSaleSapPayload.ToEventId()
+            && call.GetArgument<IEnumerable<KeyValuePair<string, object>>>(2)!.ToDictionary(c => c.Key, c => c.Value)["{OriginalFormat}"].ToString() == "The record of sale SAP payload created. | _X-Correlation-ID : {_X-Correlation-ID}").MustHaveHappenedOnceExactly();
+        }
+
+        [Test]
+        public void WhenTransactionTypeIsMigrateNewLicence_ThenReturns_SomeFieldsEmptyInSapXmlPayloadCreationTests()
```

Comment:
Test name doesn't look good. Can you update it?

Reply:
We have followed the existing naming convention for the Test Name.

Reply:
Changed to WhenTransactionTypeIsMigrateNewLicence_ThenReturns_SapXmlPayloadWithSomeEmptyFields